wradlib/georef/raster.py

```diff
@@ -474,3 +482,98 @@ def reproject_raster_dataset(src_ds, **kwargs):
     gdal.ReprojectImage(src_ds, dst_ds, src_srs, dst_srs, resample)
 
     return dst_ds
+
+
+def create_raster_dataset(data, coords, projection=None, nodata=-9999):
+    """ Create In-Memory Raster Dataset
+
+    Parameters
+    ----------
+    data : :class:`numpy:numpy.ndarray`
+        Array of shape (rows, cols) or (bands, rows, cols) containing
+        the data values.
+    coords : :class:`numpy:numpy.ndarray`
+        Array of shape (rows, cols, 2) containing xy-coordinates.
+    projection : osr object
+        Spatial reference system of the used coordinates, defaults to None.
+    nodata : int
+        Value of NODATA
+
+    Returns
+    -------
+    dataset : gdal.Dataset
+        In-Memory raster dataset
+
+    Note
+    ----
+    The origin of the provided data and coordinates is UPPER LEFT.
+    """
+
+    # align data
+    data = data.copy()
+    if data.ndim == 2:
+        data = data[np.newaxis, ...]
+    bands, rows, cols = data.shape
+
+    # create In-Memory Raster with correct dtype
+    mem_drv = gdal.GetDriverByName('MEM')
+    gdal_type = gdal_array.NumericTypeCodeToGDALTypeCode(data.dtype)
+    dataset = mem_drv.Create('', cols, rows, bands, gdal_type)
+
+    # initialize geotransform
+    x_ps, y_ps = coords[1, 1] - coords[0, 0]
+    upper_corner_x = coords[0, 0, 0] - x_ps/2
+    upper_corner_y = coords[0, 0, 1] - y_ps/2
+    geotran = [upper_corner_x, x_ps, 0, upper_corner_y, 0, y_ps]
+    dataset.SetGeoTransform(geotran)
+
+    if projection:
+        dataset.SetProjection(projection.ExportToWkt())
+
+    # set np.nan to nodata
+    dataset.GetRasterBand(1).SetNoDataValue(nodata)
+
+    for i, band in enumerate(data, start=1):
+        dataset.GetRasterBand(i).WriteArray(band)
+
+    return dataset
+
+
+def merge_rasters(datasets):
+    """Merge rasters.
+
+    Parameters
+    ----------
+    datasets : list of gdal.Dataset
+        raster images with georeferencing
+
+    Returns
+    -------
+    dataset : gdal.Dataset
+        merged raster dataset
+    """
+
+    dataset = gdal.Warp('', datasets, format='MEM')
+
+    return(dataset)
+
+
+def raster_to_polyvert(dataset):
+    """Get raster polygonal vertices from gdal dataset.
+
+    Parameters
+    ----------
+    dataset : gdal.Dataset
+        raster image with georeferencing (GeoTransform at least)
+
+    Returns
+    -------
+    polyvert : :class:`numpy:numpy.ndarray`
+        A 3-d array of polygon vertices with shape (..., 5, 2).
+
+    """
+    rastercoords = read_gdal_coordinates(dataset, mode="edges")
+
+    polyvert = util.grid_to_polyvert(rastercoords)
```

Comment:
I would rather see the function `grid_to_polyvert` in the `wradlib.georef.rect` submodule.